File: 0016-3sum-closest/0016-3sum-closest.java
class Solution {
    public int threeSumClosest(int[] nums, int target) {
        int n=nums[0]+nums[1]+nums[2];
        Arrays.sort(nums);
        for(int i=0;i<nums.length;i++){
            int l=i+1;
            int r=nums.length-1;
            while(l<r){
                int sum=nums[i]+nums[l]+nums[r];
                if(Math.abs(n-target)>Math.abs(sum-target)) n=sum;
                else if(sum<target) l++;
                else if(sum>target) r--;
                else return target;
            }
        }
        return n;
    }
}

File: 0016-3sum-closest/0016-3sum-closest.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var threeSumClosest = function(nums, target) {
    let cs=nums[0]+nums[1]+nums[2];
    nums.sort((a,b)=>a-b);
    for(let i=0;i<nums.length;i++){
        let l=i+1;
        let r=nums.length-1;
        while(l<r){
            let sum=nums[i]+nums[l]+nums[r];
            if(Math.abs(cs-target)>Math.abs(sum-target)){ 
                cs=sum;
            }
            if(sum<target)l++;
            else if(sum>target) r--;
            else return sum;
        }
    }
    return cs;
};